samples/aui/auidemo.cpp

```diff
@@ -1356,6 +1376,17 @@ void MyFrame::OnPaneClose(wxAuiManagerEvent& evt)
     }
 }
 
+// copy state from user-controllable checkbox to disabled checkbox
+void MyFrame::On3CheckUpdateUI(wxUpdateUIEvent& evt)
+{
+#if wxUSE_CHECKBOX
+    wxWindow* tb4 = m_mgr.GetPane("tb4").window;
+    wxWindow* wnd = tb4->GetWindowChild(ID_3CHECK);
+    wxCheckBox* src = wxDynamicCast(wnd, wxCheckBox);
```

Comment:
I'd use

```suggestion
    wxCheckBox* src = wxCheckCast<wxCheckBox>(wnd);
```

to at least get an assert before crashing if it ever turns out not to be a checkbox.

(here and elsewhere)

Reply:
OK

Reply:
changes force-pushed